tests/integration/components/flash-message-test.js

```diff
@@ -111,4 +112,24 @@ if (parseFloat(Ember.VERSION) > 2.0) {
 
     clock.restore();
   });
+
+  test('a custom component can use the close closure action', function(assert) {
+    assert.expect(1);
+
+    this.set('flash', FlashMessage.create({ message: 'hi', sticky: true, closeOnClick: false }));
+
+    this.render(hbs`
+      {{#flash-message flash=flash as |component flash close|}}
+        <a href="#" {{action close}}>close</a>
+        <a href="#">stay open</a>
+      {{/flash-message}}
+    `);
+    // this.$(":contains(stay open)").click();
+
+    this.$(":contains(close)").click();
+    return wait().then(() => {
+      assert.equal(this.$().text().trim(), '');
```

Comment:
Hey @caseywatts. Thanks for the PR!

Rather than testing that the flash message itself disappears, you can stub the `destroyMessage` call on the flash-object with sinon, like we do here https://github.com/caseywatts/ember-cli-flash/blob/master/tests/integration/components/flash-message-test.js#L50

Then, you can test that it is `calledOnce` here like we do here:
https://github.com/caseywatts/ember-cli-flash/blob/master/tests/integration/components/flash-message-test.js#L69

Then we aren't testing that the flash message disappears, but that clicking `close` communicates with the flash-objects API in the way we expect.

Then maybe we can test the logic of it no longer rendering in an acceptance test 😄